### PR TITLE
Add postgresql memory store

### LINKED ISSUE #166: Use PostgresSQL as MemoryStore
Current Memory store options are: 
- volatile
- qdrant
- azure cognitive search
- chroma

While deploying to Azure, the only PaaS available directly on Azure is Azure Cognitive Search. 
However this is limited to 50 indexes, that might limit the usage when adding more users to the chat...

I'ld like to add CosmosDB as vector database (using PostgreSQL connector).

### PR DESCRIPTION
**Pull request by kbeaugrand**
### Motivation and Context

I would like to provide a production-grade PaaS solution to store the vector database.
It provides PostgreSQL database support for memory. We can now use Azure Database for PostgreSQL or CosmosDB for PostgreSQL.

Fixes #166 

### Description

I added configuration support for PostgreSQL. in the kernel initialization. 
Furthermore I updated the deployment script to support this configuration.

> Note: In the deployment scripts, I've added the SqlAdminPassword parameter, which must be supplied during provisioning. I intentionally named it without the PostgreSQL name, as I also worked on implementing SQL Server Memory store in the SK repository. This parameter could be used by both databases during deployment.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
> Not found test project on the API part ;(
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: scripts/deploy/deploy-azure.ps1 [hunk 1]
```diff
@@ -44,11 +44,15 @@ param(
...
 
-    [ValidateSet("Volatile", "AzureCognitiveSearch", "Qdrant")]
+    [ValidateSet("Volatile", "AzureCognitiveSearch", "Qdrant", "Postgres")]
     [string]
     # What method to use to persist embeddings
     $MemoryStore = "AzureCognitiveSearch",
 
+    [SecureString]
+    # Password for the Postgres database
+    $SqlAdminPassword = '',
```
Comment (glahaye) [inline scripts/deploy/deploy-azure.ps1:54]: Nit:

```suggestion
    $SqlAdminPassword = "",
```

File: scripts/deploy/deploy-azure.ps1 [hunk 2]
```diff
@@ -87,6 +91,10 @@ if ($AIService -eq "OpenAI" -and !$AIApiKey) {
     exit 1
 }
 
+if ($MemoryStore -eq "Postgres" -and !$SqlAdminPassword) {
+    exit 1
```
Comment (glahaye) [inline scripts/deploy/deploy-azure.ps1:95]: You should print a message saying why we are exiting.

File: scripts/deploy/main.bicep [hunk 3]
```diff
@@ -54,6 +54,7 @@ param deployCosmosDB bool = true
   'Volatile'
```
Comment (glahaye) [inline scripts/deploy/main.bicep:54]: Please update the corresponding ARM JSON file (main.json) to reflect your changes. The "Deploy to Azure" button only works with JSON ARM templates.

Command to use:  az bicep build --file main.bicep

File: webapi/Extensions/SemanticKernelExtensions.cs [hunk 4]
```diff
@@ -220,6 +223,24 @@ private static void AddSemanticTextMemory(this IServiceCollection services)
                     );
                 });
                 break;
+            case MemoryStoreOptions.MemoryStoreType.Postgres:
```
Comment (glahaye) [inline webapi/Extensions/SemanticKernelExtensions.cs:226]: Nit: please a line between the previous "break" and this case.

**Comment by kbeaugrand** [comment]
@glahaye thank you for the review, I apply fixes from your suggestions and comments.

**Review by glahaye — APPROVED** [review]
LGTM

**Pull request merged**